File: client/src/types/notification.ts
export type NotificationType = 1 | 2 | 3 | 4;

export type AppNotification = {
  id: number;
  type: NotificationType;
  title: string;
  body: string;
  timeTaskId?: number;
  groupId?: number;
  conversationId?: number;
  createdAtUtc: string;
  readAtUtc?: string;
  isRead: boolean;
};

File: API/Entities/Notification.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace API.Entities;

public class Notification
{
    public int Id { get; set; }
    public required string RecipientMemberId { get; set; }

    [JsonIgnore]
    public Member RecipientMember { get; set; } = null!;

    public NotificationType Type { get; set; }

    [MaxLength(120)]
    public required string Title { get; set; }

    [MaxLength(300)]
    public required string Body { get; set; }

    public int? TimeTaskId { get; set; }
    public int? GroupId { get; set; }
    public int? ConversationId { get; set; }
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? ReadAtUtc { get; set; }
}

public enum NotificationType
{
    TaskApplicationReceived = 1,
    TaskApplicationAccepted = 2,
    TaskCompleted = 3,
    TaskCancelled = 4
}
